pkg/comp-functions/functions/vshnredis/script/cleanupRestore.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -euo pipefail
+
+echo "Wait for restore to complete"
+
+until [ "$(kubectl -n "${TARGET_NAMESPACE}" get job "${RESTORE_JOB_NAME}" -o jsonpath='{.status.succeeded}')" -eq 1 ];
```

Comment:
This will not go into infinite loop in case the restore job failed right?

Reply:
It actually will. Not sure if we should handle this differently.

Reply:
> Maybe it makes sense to use `kubectl wait --for=condition=succeeded --timeout=300s` or something?

This will work if the resource is already created by the time kubectl wait is issued. If that's the case then we should implement this solution.

Reply:
> This will work if the resource is already created by the time kubectl wait is issued. If that's the case then we should implement this solution.

We can't guarantee that. It really depends if the provider-helm is faster in deploying the release then the composite function is in deploying the job.

Reply:
I would just add a timeout with bash in this case.

Reply:
It's honestly not clear to me how we should handle a restore failure.  Just scaling back up, even through we did not restore isn't really the correct behavior.

Maybe we'd need some way to bubble up the failure information to the user.

Reply:
We could maybe add a "timeout" to this script with something like:
```
counter=0
until [ $counter -eq 300 ] ||  kubectl -n ...
do
  sleep 1
  $((counter++))
done
```

This should exit after 5 minutes regardless of the object's status.

Reply:
Good idea, added a counter variable and adjusted the `until` loop accordingly

Reply:
I have added the counter and adjusted the scripts a bit to not "leak" command outputs where it's not useful and made the loops more robust as well.

Reply:
Should be fine for now. But I guess we sooner or later run into instances that won't complete the restore within 5 minutes due to their size.

Reply:
Hm didn't consider that part. I'll think about this a little bit longer and try to come up with a more resilient approach, as the 5 minute limit for restore is problematic.

Reply:
I refactored that part a bit. Please check again if that solution is acceptable.